gtk/gtkwindow.c

```diff
@@ -673,6 +673,45 @@ startup_id_is_fake (const gchar* startup_id)
   return strncmp (startup_id, "_TIME", 5) == 0;
 }
 
+#ifdef GDK_WINDOWING_X11
+gboolean
+gtk_window_property_notify (GtkWidget *widget, GdkEventProperty *event)
+{
+  static GdkAtom unredirected;
+
+  g_return_val_if_fail (widget != NULL, FALSE);
+  g_return_val_if_fail (event != NULL, FALSE);
+
+  if (g_once_init_enter (&unredirected))
```

Comment:
There's really no need to use GOnce, here: GDK is not thread safe.

All atoms, as far as I remember, are initialised when the display is opened, so you'll have to store this atom on the `GdkDisplay` instance and retrieve it here.